### ISSUE TITLE
Work nicely together with `commit --verbose`

### ISSUE THREAD

**Issue by arnoFleming** [tombruijn/lintje#21]
I often use `git commit --verbose` to have a last check to not commit rubbish (or at least unintentional rubbish).

Lintje is assessing the lines after the magic comment line as if they're part of the commit message.

Notice that the lines underneath line 17 in the example, do not actually make it into the commit, they're meant as 'FYI these are the changes you're committing'. The lines themselves aren't commented out, they're just below the magic comment line

## Steps to reproduce: 

1. Create a commit with long lines in the code changes
2. Commit the change from the cli with `EDITOR=vim commit -v` (other editors may break/work, but I want you to be able to reproduce for sure)
3. run `lintje` on the commit (in the example it was run from a non-breaking hook according to the README)

## Example output from a commit

The commit window is as follows:

```text
  1 Let's mention our streaming sponsor on our website
  2
  3 We have the streaming partner sponsor package, this amends that work to
  4 also mention our sponsor on our amsrb.org page. May (but prolly will
  5 not) boost their SEO ranking (but will deffo boost their ego 😇 )
  6
  7 # Please enter the commit message for your changes. Lines starting
  8 # with '#' will be ignored, and an empty message aborts the commit.
  9 #
 10 # On branch streaming-sponsor
 11 # Your branch is behind 'origin/streaming-sponsor' by 1 commit, and can be fast-forwarded.
 12 #   (use "git pull" to update your local branch)
 13 #
 14 # Changes to be committed:
 15 #┊      modified:   sponsorships/index.html
 16 #
 17 # ------------------------ >8 ------------------------
 18 # Do not modify or remove the line above.
 19 # Everything below it will be ignored.
 20 diff --git a/sponsorships/index.html b/sponsorships/index.html
 21 index 3d21fb0..03d32f7 100644
 22 --- a/sponsorships/index.html
 23 +++ b/sponsorships/index.html
 24 @@ -13,19 +13,20 @@ title: Sponsorships
 25    <p>All sponsors are required to read and adhere to our <a href="https://amsrb.org/code-of-conduct/">Code of Condu
 26
 27    <h2>Streaming partner - € 250</h2>
 28 -
 29 +
 30    <p>As we're looking at virtual-only events a <em>little</em> while longer, we're looking for parties that'd like
 31 -
 32 +
 33    <p>In return you'll receive:</p>
 34
 35    <ul>
 36      <li>Mention in the general channel on Slack</li>
 37      <li>Mention on meetup.com</li>
 38 +    <li>Mention on amsrb.org</li>
 39      <li>1 slide, and a shoutout during the meetup’s intro</li>
 40      <li>Optional: instead of the slide, you can submit a 1 minute-long, recorded pitch</li>
 41      <li>(@)mention on Twitter</li>
 42    </ul>
 43 -
 44 +
 45    <h2>Location sponsor</h2>
 46
 47    <p>Hosting an Amsterdam.rb means your location is (wheelchair) accessible, and you have the following on-site:</p
~
```

When I exit the commit message editor (saving the message for commit), Lintje responds with raised eyebrows:
```text
MessageLineLength: Line 17 in the message body is longer than 72 characters
  0000000:17:73: Let's mention our streaming sponsor on our website
     |
  17 |    <p>As we're looking at virtual-only events a <em>little</em> while longer, we're looking for parties that'd like to help us cover our costs.</p>
     |                                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Shorten line to maximum 72 characters

MessageLineLength: Line 27 in the message body is longer than 72 characters
  0000000:27:73: Let's mention our streaming sponsor on our website
     |
  27 |      <li>Optional: instead of the slide, you can submit a 1 minute-long, recorded pitch</li>
     |                                                                         ^^^^^^^^^^^^^^^^^^^^ Shorten line to maximum 72 characters

MessageLineLength: Line 34 in the message body is longer than 72 characters
  0000000:34:73: Let's mention our streaming sponsor on our website
     |
  34 |    <p>Hosting an Amsterdam.rb means your location is (wheelchair) accessible, and you have the following on-site:</p>
     |                                                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Shorten line to maximum 72 characters

1 commit and branch inspected, 3 violations detected

Lintje failure
[streaming-sponsor d609231] Let's mention our streaming sponsor on our website
 1 file changed, 4 insertions(+), 3 deletions(-)
```

**Comment by tombruijn** [comment]
Hi @arnoFleming!

That looks like a bug yes, but I can't reproduce.

I assume you use the default `commit.cleanup` mode? (The command `git config commit.cleanup` outputs nothing.) Then I can reproduce it.

What happens is that `git commit -v` prints a scissor line, which Lintje doesn't interpret as a scissor line. It only does that in `commit.cleanup` mode `scissor`.

I'll have to revisit how to handle this for the verbose mode in combination with the default cleanup mode. Which doesn't really seem to match up with the [cleanup docs](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---cleanupltmodegt). The strip mode doesn't mention the scissor line and its behavior for that.

> default: Same as strip if the message is to be edited. Otherwise whitespace.
> strip: Strip leading and trailing empty lines, trailing whitespace, commentary and collapse consecutive empty lines.

**Comment by arnoFleming** [comment]
> That looks like a bug!

Well, one persons' feature is another persons' bug, right? Happy to know we're on the same page though...

`git config commit.cleanup` does indeed have an empty output, so that would be the default

I would be happy if I could configure this behaviour with a cli flag. Dunno how many peeps use `commit --verbose` in the first place. If that makes life simpler on your end, I'd be happy to change how Lintje is invoked in the git hook.

PS: In case my git version matters: It's *2.34.0*

**Comment by tombruijn** [comment]
@arnoFleming I fixed the issue you reported in [this commit](https://github.com/tombruijn/lintje-private/commit/75271f5e47c708373b28f93fd868a791f857d779). It will now ignore everything below the scissor line in every cleanup mode. Apparently the Git commit docs don't describe exactly how these cleanup modes work, because all modes scan for the scissors line.

While I was reviewing the implementation I fixed two other issues with the parsing of the commit message hook file as well.

- [Strip trailing whitespace in scissors cleanup mode](https://github.com/tombruijn/lintje-private/commit/d85243e40b3216cb671ff4663e9b4961a8844f4e)
- [Improve leading empty line detection](https://github.com/tombruijn/lintje-private/commit/7daeab7e49517910aab3c21e3ace84f68aef71a6)
